docs/references/config/config.toml.md

```diff
@@ -0,0 +1,1997 @@
+```toml
+tablename.property_key = "default value"
+```
+
+| Value type          | string/bool/int/etc                | <- can be further clarified (duration, hex, base64-encoded)
+|:--------------------|:-----------------------------------|
+| **Possible values** | `"default value or exact example"` |
+|                     | `generic description`              |
+
+Lorem ipsum and everything else that you can think about this value and stuff.
+This is the enhanced description and additional notes that is worth talking about.
+Description about the possible values too.
+--->
+
+# config.toml
+The `config.toml` file is a standard [TOML](https://toml.io/en/v1.0.0) file that configures the basic functionality
+of CometBFT, including the configuration of the reactors.
+
+All relative paths in the configuration are relative to `$CMTHOME`.
+(See [the HOME folder](./README.md#the-home-folder) for more details.)
+
+## Base configuration
+The root table defines generic node settings. It is implemented in a struct called `BaseConfig`, hence the name.
+
+### version
+The version of the CometBFT binary that created or last modified the config file.
+```toml
+version = "1.0.0"
+```
+
+| Value type          | string                  |
+|:--------------------|:------------------------|
+| **Possible values** | semantic version string |
+|                     | `""`                    |
+
+This string validates the configuration file for the binary. The string has to be either a
+[valid semver](https://semver.org) string or an empty string. In any other case, the binary halts with an
+`ERROR: error in config file: invalid version string` error.
+
+In the future, the code might make restrictions on what version of the file is compatible with what version of the
+binary. There is no such check in place right now. Configuration and binary versions are interchangeable.
+
+### proxy_app
+The TCP or UNIX socket of the ABCI application or the name of an example ABCI application compiled in with the CometBFT
+library.
+```toml
+proxy_app = "tcp://127.0.0.1:26658"
+```
+
+| Value type          | string                                                  |
+|:--------------------|:--------------------------------------------------------|
+| **Possible values** | TCP Stream socket (e.g. `"tcp://127.0.0.1:26658"`)      |
+|                     | Unix domain socket (e.g. `"unix:///var/run/abci.sock"`) |
+|                     | `"kvstore"`                                             |
+|                     | `"persistent_kvstore"`                                  |
+|                     | `"noop"`                                                |
+
+When the ABCI application is written in a different language than Golang, (for example the
+[Nomic binary](https://github.com/nomic-io/nomic) is written in Rust) the application can open a TCP port or create a
+UNIX domain socket to communicate with CometBFT, while CometBFT runs as a separate process.
+
```

Comment:
I would add here a couple of sentences here to say that the "TCP Stream socket" can take different ports. Also, that using an IP address different from `localhost` (IPv4: `127.0.0.1`, IPv6: `::1`) is strongly discouraged (because it has strong performance and security implications)